### DISCUSSION TITLE
Audio records deleted after update

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#518] [Q&A] [answered]
Hi,

Last night I updated my Birdnet-Pi and all the audio records were deleted, even those I selected as non to be purged.

What could be the reason?

Thank you.

**Comment by mcguirepr89** [comment, marked as answer]
@blogdemoi I'm very sorry to hear that. The only thing I can think that would cause that is if by accident "Clear ALL Data" were selected instead of "Update."
![Screenshot from 2022-08-26 08-47-50](https://user-images.githubusercontent.com/60325264/186906806-7b192bcf-69d7-4451-8f25-da5275d97ffe.png)

@ehpersonal38 Anything new that might have overwritten the database _and_ removed all sound files?
(My system is many many updates behind)

**Reply by ehpersonal38** [reply]
I'd have to concur with Patrick here, a simple update shouldn't have done anything to remove your audio files. @blogdemoi Can you confirm if your database was wiped or if it was just the recordings? If the DB was wiped then your "total" number of detections on the Overview page will have decreased drastically.

**Reply by blogdemoi** [reply]
Hi,

My database is fine. What I don't understand is why the records that were selected as non deletable were deleted.

I am not at home now, so I will double check everything this evening (French time here). It is not very important anyway,my Birdnet-Pi has only been running for one week. I am more concerned about the recordings marked as "do not delete".

**Reply by ehpersonal38** [reply]
Files excluded from purge will have their wav file path written to "$HOME/scripts/disk_check_exclude.txt". Can you verify that this file has text entries in it?

**Comment by blogdemoi** [comment]
> Files excluded from purge will have their wav file path written to "$HOME/scripts/disk_check_exclude.txt". Can you verify that this file has text entries in it?

I don't see any wav file path.

Entries:
![3](https://user-images.githubusercontent.com/83965802/186988145-b15d9b83-813f-4d4f-b3ee-4d42227d633a.png)


Weird things, it says no detections for today (red arrow) but I do have detections:
![1](https://user-images.githubusercontent.com/83965802/186987262-bdbd0fd0-7427-4f43-8743-f706f10f4f17.png)

> If the DB was wiped then your "total" number of detections on the Overview page will have decreased drastically.

My database seems fine as I have 3646 detections (over a six days period)
![2](https://user-images.githubusercontent.com/83965802/186987816-73f2694b-d04a-4ec4-b946-076557e643b8.png)

**Reply by mcguirepr89** [reply]
Hmmm I'll have to perform an update soon and see if I can replicate this. Looks like the links to the detections are not pointing to files, but analyses are being written to the database.

I will let you know if I am able to replicate the issue and we'll see where to go from there. :crossed_fingers:

**Reply by mcguirepr89** [reply]
Not able to replicate anything like that with an update (thank goodness!!)

Will you please send the output from this command? 
_Note:If it is around 3,646+ lines long, don't send the output and instead just let me know that all the files are there._ 
```
find ~/BirdSongs/Extracted | sed "s|$HOME||"
```

**Reply by blogdemoi** [reply]
```
birdnet:~/BirdNET-Pi$ find ~/BirdSongs/Extracted | sed "s|$HOME||"
/BirdSongs/Extracted
/BirdSongs/Extracted/spectrogram.png
/BirdSongs/Extracted/phpsysinfo
/BirdSongs/Extracted/weekly_report.php
/BirdSongs/Extracted/history.php
/BirdSongs/Extracted/play.php
/BirdSongs/Extracted/Charts
/BirdSongs/Extracted/Charts/Combo-2022-08-22.png
/BirdSongs/Extracted/Charts/Combo2-2022-08-19.png
/BirdSongs/Extracted/Charts/Combo-2022-08-26.png
/BirdSongs/Extracted/Charts/Combo2-2022-08-23.png
/BirdSongs/Extracted/Charts/Combo-2022-08-21.png
/BirdSongs/Extracted/Charts/Combo-2022-08-18.png
/BirdSongs/Extracted/Charts/Combo2-2022-08-22.png
/BirdSongs/Extracted/Charts/Combo2-2022-08-18.png
/BirdSongs/Extracted/Charts/Combo-2022-08-24.png
/BirdSongs/Extracted/Charts/Combo-2022-08-19.png
/BirdSongs/Extracted/Charts/Combo2-2022-08-26.png
/BirdSongs/Extracted/Charts/Combo2-2022-08-21.png
/BirdSongs/Extracted/Charts/Combo-2022-08-25.png
/BirdSongs/Extracted/Charts/Combo2-2022-08-20.png
/BirdSongs/Extracted/Charts/Combo-2022-08-20.png
/BirdSongs/Extracted/Charts/Combo-2022-08-23.png
/BirdSongs/Extracted/Charts/Combo2-2022-08-24.png
/BirdSongs/Extracted/Charts/Combo2-2022-08-25.png
/BirdSongs/Extracted/overview.php
/BirdSongs/Extracted/images
/BirdSongs/Extracted/stats.php
/BirdSongs/Extracted/style.css
/BirdSongs/Extracted/scripts
/BirdSongs/Extracted/todays_detections.php
/BirdSongs/Extracted/Processed
/BirdSongs/Extracted/favicon.ico
/BirdSongs/Extracted/index.php
/BirdSongs/Extracted/views.php
/BirdSongs/Extracted/spectrogram.php
/BirdSongs/Extracted/static
birdnet:~/BirdNET-Pi$
```

**Reply by blogdemoi** [reply]
I am re-installing from scratch. As my Birdnet-Pi is only one week old I prefer to do everything again, then I will do an image of the sd card using dd every week in case things go bad.

**Reply by blogdemoi** [reply]
Hello, I am back online, a new sd card, bigger size, a fresh install and after much fighting with my reverse proxy it seems I am online again. As I have a small usb key attached to this Pi, is it possible to mount it and have the audio records on it?

Thank you.

**Reply by mcguirepr89** [reply]
@blogdemoi That's great! I'm glad you were able to figure it out.

> As I have a small USB key attached to this Pi, is it possible to mount it and have the audio records on it?

You can definitely do this, but there are a few things to keep in mind:
- The file system type for BirdNET-Pi is `ext4` -- Windows cannot natively handle the `ext4` filesystem, so you'll need another program on your Windows machine if your goal is to move the files from the BirdNET-Pi over to a Windows PC/Laptop.
- You'll need to make sure that the USB is _always_ mounted while the services are running so that they don't break. Basically, if you configure the system to put files on the USB key, any service that cannot find the USB's mounted drive will fail and could have some really shitty, unexpected consequences. So be sure to "Tools" > "Services" > "Stop Core Services" before configuring the USB mount and to "Restart Core Services" after the USB is mounted. Likewise, be certain to "Stop Core Services" anytime you plan to remove the USB key for any reason.

To accomplish what your question wondered:
1. Clear out the USB and format it as `ext4` (you can use the Raspberry Pi Imager tool for this)
2. Plug it into the Pi
3. It _should_ have automatically mounted in the `/media/$USER` directory.
4. (Optional) Create an "Extracted" directory in the `/media/$USER` directory where you will put identified bird sounds
5. Edit your `birdnet.conf` file so that the "`EXTRACTED`" variable is configured along the lines of the example below, using `birder` as the `$USER`
   ```
   EXTRACTED=/media/birder/Extracted
   ```

The steps above will tell the system to write its "Extracted" detections to the `/media/birder/Extracted` directory which is the USB key.

Best of luck setting this up. For what it is worth, you may find using the "Tools" > "File Manager" a bit easier for managing the files than physically transferring the files.

My very best regards as always,
Patrick

**Reply by blogdemoi** [reply]
Thank you, I will try to do that.

> so you'll need another program on your Windows machine if your goal is to move the files from the BirdNET-Pi over to a Windows PC/Laptop.

The last time I used Windows was 10 years ago, I go all Linux now :-)

**Reply by mcguirepr89** [reply]
>The last time I used Windows was 10 years ago, I go all Linux now :-)

Then you're in good company -- You might consider setting up an rsync service since you're on Linux -- you could just sync the most recents, or sync each previous day's detections. Lot's of wiggle room knowing that you're on Linux for your PC. You may instead mount your Linux laptop as an SSHFS onto the BirdNET-Pi, then the BirdNET-Pi could literally write its detections to your laptop/PC :)
